app/assets/javascripts/discourse/app/components/route-control.js

```diff
@@ -0,0 +1,51 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import { defaultHomepage } from "discourse/lib/utilities";
+
+export default class RouteControl extends Component {
+  @service router;
+  @service siteSettings;
+  @service currentUser;
+
+  get canDisplay() {
+    const currentRouteName = this.router.currentRouteName;
```

Comment:
When the getter accesses things like `router.currentRouteName`, it adds that property to the autotracking context. With the current implementation, that means the entire getter will be evaluated every time `currentRouteName` changes, even if `currentRouteName` is irrelevant to the result.

To improve this, we can defer looking up `currentRouteName` until we actually need it. (i.e. move line 11 down to line 28. 

(granted, this is a tiny optimisation, but if we're hoping for this to be used in lots of places then it's worth thinking about)